src/h3lib/lib/faceijk.c

```diff
@@ -886,8 +901,26 @@ int _adjustOverageClassII(FaceIJK* fijk, int res, int pentLeading4,
 
         // overage points on pentagon boundaries can end up on edges
         if (substrate && ijk->i + ijk->j + ijk->k == maxDim)  // on edge
-            overage = 1;
+            overage = FACE_EDGE;
     }
 
     return overage;
 }
+
+/**
+ * Adjusts a FaceIJK address for a pentagon vertex in a substrate grid in
+ * place so that the resulting cell address is relative to the correct
+ * icosahedral face.
+ *
+ * @param fijk The FaceIJK address of the cell.
+ * @param res The H3 resolution of the cell.
+ */
+Overage _adjustPentVertOverage(FaceIJK* fijk, int res) {
+    int pentLeading4 = 0;
+    Overage overage = _adjustOverageClassII(fijk, res, pentLeading4, 1);
+    while (overage == NEW_FACE) {
```

Comment:
nit: Any preference for or against a do/while loop?
```
Overage overage;
do {
    overage = _adjustOverageClassII(fijk, res, pentLeading4, 1);
} while(overage == NEW_FACE);
```

Reply:
Good point, that's a bit cleaner. Will update.